riptide-spring-boot-autoconfigure/src/main/java/org/zalando/riptide/autoconfigure/DefaultRiptideRegistrar.java

```diff
@@ -407,11 +407,19 @@ private Optional<String> registerCircuitBreakerFailsafePlugin(final String id, f
         if (client.getCircuitBreaker().getEnabled()) {
             final String pluginId = registry.registerIfAbsent(name(id, CircuitBreaker.class, FailsafePlugin.class),
                     () -> {
+                        var executorService = registry.find(name(id, "CircuitBreaker", ExecutorService.class));
+
                         log.debug("Client [{}]: Registering [CircuitBreakerFailsafePlugin]", id);
-                        return genericBeanDefinition(FailsafePluginFactory.class)
-                                .setFactoryMethod("createCircuitBreakerPlugin")
-                                .addConstructorArgValue(registerCircuitBreaker(id, client))
-                                .addConstructorArgValue(createTaskDecorators(id, client));
+                        return executorService.map(executorServiceBeanName -> genericBeanDefinition(FailsafePluginFactory.class)
```

Comment:
Given that `registry.find` gives you an optional, can't you just use `map(name -> ref(name)).orElse(null)` and then just pass this to the last `addConstructorArg` call given that you nevertheless allow to call it with `null`? Then you only need to call the `genericBeanDefiniton` once.